### PR TITLE
Link libmount less

### PR DESCRIPTION
**Pull request by keszybz**
This does some rearranging of code, which results in a huge difference in linkage. Somehow, `--as-needed` suddenly works.

`nss_*.so.2` (with the exception of nss-myhostname) are still linked to libcap. But it's a tiny library and as long as libsystemd-bus uses it, we won't be able to get rid of it.

### CODE REVIEW

File: src/shared/mount-util.h [hunk 1]
```diff
@@ -1,25 +1,12 @@
 /* SPDX-License-Identifier: LGPL-2.1+ */
 #pragma once
 
-#include <fcntl.h>
-#include <mntent.h>
-#include <stdbool.h>
 #include <stdio.h>
-#include <sys/stat.h>
-#include <sys/types.h>
+#include <mntent.h>
```
Comment (yuwata) [inline src/shared/mount-util.h:4]: This unsorted inclusion is intended?

Comment (keszybz) [reply src/shared/mount-util.h:4]: Nope. I'll fix it once the CI passes.

**Comment by poettering** [comment]
needs rebase...

**Comment by poettering** [comment]
also 2 of the ubuntu CIs passed, i figure that's good enough

**Comment by keszybz** [comment]
Rebased. The only change is the header sort order fix.

**Pull request merged**